src/attributes/descriptors.js

```diff
@@ -11,7 +11,8 @@ exports.attributeDescriptorFor = function attributeDescriptorFor(attributeName,
     },
 
     set(value) {
-      this.attributes[attributeName] = schema[attributeName].coerce(value);
+      if(schema[SCHEMA_OPTIONS].coerceAttributes) this.attributes[attributeName] = schema[attributeName].coerce(value);
```

Comment:
Avoid inline `if`/`else` like this, it's not good for readability and can cause problems when someone changes this file because of typos.
```suggestion
      if(schema[SCHEMA_OPTIONS].coerceAttributes) {
        this.attributes[attributeName] = schema[attributeName].coerce(value);
      }
```